src/com/maddyhome/idea/vim/command/CommandState.java

```diff
@@ -86,9 +88,22 @@ public static boolean inVisualCharacterMode(@Nullable Editor editor) {
     return state.getMode() == Mode.VISUAL && state.getSubMode() == SubMode.VISUAL_CHARACTER;
   }
 
+  public static boolean inVisualMode(@Nullable Editor editor) {
+    return getInstance(editor).getMode() == Mode.VISUAL;
+  }
+
+  public static boolean inSelectMode(@Nullable Editor editor) {
+    return getInstance(editor).getMode() == Mode.SELECT;
+  }
+
+  public static boolean inVisualLineMode(@Nullable Editor editor) {
+    final CommandState state = getInstance(editor);
+    return state.getMode() == Mode.VISUAL && state.getSubMode() == SubMode.VISUAL_LINE;
+  }
+
   public static boolean inVisualBlockMode(@Nullable Editor editor) {
     final CommandState state = getInstance(editor);
-    return state.getMode() == Mode.VISUAL && state.getSubMode() == SubMode.VISUAL_BLOCK;
+    return state.getSubMode() == SubMode.VISUAL_BLOCK;
```

Comment:
It makes sense to check if the main mode is still visual. I see no reason to drop this check.